server/main.rs

```diff
@@ -1,20 +1,59 @@
 extern crate hematite_server as hem;
+#[macro_use]
+extern crate log;
 
-use std::io;
+use std::net::TcpListener;
+use std::sync::Arc;
+use std::thread;
 
-use hem::packet::Protocol;
+use hem::vanilla::Server;
 
-fn main() {
-    let value = false;
-    let first_value = vec![0];
+use log::{LogLevel, LogLevelFilter, LogMetadata, LogRecord, SetLoggerError};
 
-    let mut w = Vec::new();
-    <bool as Protocol>::proto_encode(&value, &mut w).unwrap();
-    assert_eq!(&w, &first_value);
+struct SimpleLogger;
 
-    let mut r = io::Cursor::new(w);
-    let value = <bool as Protocol>::proto_decode(&mut r).unwrap();
-    assert_eq!(false, value);
+impl log::Log for SimpleLogger {
+    fn enabled(&self, metadata: &LogMetadata) -> bool {
+        metadata.level() <= LogLevel::Info
+    }
 
-    println!("It works!");
+    fn log(&self, record: &LogRecord) {
+        if self.enabled(record.metadata()) {
+            println!("{} - {}", record.level(), record.args());
+        }
+    }
+}
+
+fn init() -> Result<(), SetLoggerError> {
+    log::set_logger(|max_log_level| {
+        max_log_level.set(LogLevelFilter::Info);
+        Box::new(SimpleLogger)
+    })
+}
+
+fn main () {
+    init().unwrap();
```

Comment:
This should have a more descriptive name. `init_logger` or something. And all of the `unwrap` calls should be `expect` instead.